### ISSUE TITLE
global config file

### ISSUE THREAD

**Issue by jaymzh** [Foodcritic/foodcritic#704]
foodcritic has the ability pass in .foodcritic files in a cookbook itself, but not globally.

It would be awesome if you could pass in a --config or something with rules to disable globally.

**Comment by tas50** [comment]
I was looking into doing this last week. I started by just modifying the current .foodcritic logic to look 1 directory up and also in your home directory. That would allow per user global exclusions and it would also allow you to have a single .foodcritic file in a mono repo. The more I dug into things though I found more and more code that just isn't designed for the config concept. How we determine files and tags is fundamentally flawed. Example: We look for the .foodcritic file for every file we examine so if you have 3 potential locations that's 3 file ops for every file. I started looking at what it would take to fix some of those problems and it quickly became a much more complex problem. I think allowing the config would be a good stop gap measure because it wouldn't triple the file ops on the run.

TLDR: This would probably be a good thing to add

**Comment by mattlqx** [comment]
I would also love it if .foodcritic wasn’t ignored if -t flags are used.

Also for sake of argument, why couldn’t the whole tree upward be traversed looking for the most specific .foodcritic until it finds one or hits root?

**Comment by tas50** [comment]
The traversal problem is just an issue of disk IO. The way Foodcritic runs right now you'd be doing hundreds of checks for the file for a mid sized cookbook depending on how deeply nested the cookbook is. I often run Foodcritic against very large mono repos. That cost potentially take down a system just looking for the .foodcritic file. Could we fix that? Totally. It's just a bit harder than it appears from the outside due to a lot of tech debt in the codebase.

**Comment by Raboo** [comment]
I agree with @mattlqx.
It makes no sense in ignoring the .foodcritic file when `-t` flags are used.
